src/Hazelcast.Net/Serialization/Compact/CompactSerializationHook.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#nullable enable
+
+using System;
+
+namespace Hazelcast.Serialization.Compact
+{
```

Comment:
I couldn't understand aim of the class.

Reply:
It looks like testing purposes but wouldn't be better if we move it to Hazelcast.Testing?

Reply:
This class (the `CompactSerializationHook`) is referenced in `HazelcastClientFactory` method `CreateSerializationService` where it's added as a hook to the serialization configuration, with the following comment:
```
// add a hook that constructs and provides the compact IIdentifiedDataSerialization factory
// because some basic compact functionality such as schema distribution don't use compact
// serialization themselves.
```

Reply:
Got it!, thanks.